packages/ihe-gateway-sdk/src/client/ihe-gateway.ts

```diff
@@ -1,41 +1,128 @@
+   *
+   * @param xcpdIti55Request An array of patient discovery transaction requests.
+   * @param xcpdIti55Request[].id Unique ID for the request.
+   * @param xcpdIti55Request[].cxId The customer ID.
+   * @param xcpdIti55Request[].timestamp The timestamp of the request.
+   * @param xcpdIti55Request[].xcpdGateways The OIDs and XCPD ITI-55 URLs of each organization to make a request to.
+   * @param xcpdIti55Request[].principalCareProviderNPIs The list of NPIs of the practitioners associated with the patient.
+   * @param xcpdIti55Request[].samlAttributes The SAML attributes of the org making the request.
+   * @param xcpdIti55Request[].patient The patient data in FHIR R4 format.
+   *
+   * @returns 202 Accepted if the request was successful.
+   */
+  async getPatient(xcpdIti55Request: XCPDRequest[]): Promise<void> {
+    return await this.api.post(IHEGateway.XCPD_ENDPOINT, xcpdIti55Request);
+  }
+
   /**
-   * An XCPD request interface
-   * @param patient The patient data in FHIR R4 format.
-   * @param cxId The customer ID.
-   * @param xcpdGateways The OIDs and XCPD ITI-55 URLs of each organization to make a request to.
-   * @param principalCareProviderNPIs The list of NPIs of the practitioners associated with the patient.
-   * @param requestId Optional. Unique ID for the request. If not provided, one will be created.
+   * Query Documents (XCA-ITI-38) request.
+   * https://profiles.ihe.net/ITI/TF/Volume2/ITI-38.html
    *
-   * @returns an XCPD request to be used with an IHE Gateway.
+   * @param xcaIti38Request An array of document query transaction requests.
+   * @param xcaIti38Request[].id Unique ID for the request.
+   * @param xcaIti38Request[].cxId The customer ID.
+   * @param xcaIti38Request[].timestamp The timestamp of the request.
+   * @param xcaIti38Request[].homeCommunityId The OID of the organization to make a request to.
+   * @param xcaIti38Request[].xcpdPatientId The patient identifier.
+   * @param xcaIti38Request[].xcpdGateway The XCA ITI-38 endpoint of the organization to make a request to.
+   * @param xcaIti38Request[].samlAttributes The SAML attributes of the org making the request.
+   * @param xcaIti38Request[].classCode Optional. The class code of the document.
+   * @param xcaIti38Request[].practiceSettingCode Optional. The practice setting code of the document.
+   * @param xcaIti38Request[].facilityTypeCode Optional. The facility type code of the document.
+   * @param xcaIti38Request[].documentCreationDate Optional. The document creation date.
+   * @param xcaIti38Request[].serviceDate Optional. The service date.
    *
-   * @throws {@link ZodError}
-   * Thrown if organization OIDs or principalCareProviderNPIs don't meet their respective criteria.
+   * @returns 202 Accepted if the request was successful.
    */
+  async getDocuments(xcaIti38Request: XCA_ITI_38Request): Promise<void> {
+    return await this.api.post(IHEGateway.XCA_ITI_38_ENDPOINT, xcaIti38Request);
```

Comment:
Are we not using the schemas to validate the responses?

Not saying we should, but if we don't then let's define types with pure TS and not Zod. Or have Zod schemas to validate the responses.

Reply:
There are no responses, Mirth will simply respond to this request with a 200. The response schemas that are created are so we can handle to payload that gets sent to the api

Reply:
> The response schemas that are created are so we can handle to payload that gets sent to the api

What do you mean with "handle (the) payload"? Do we process a payload with this Zod schema? If so, can you point me to it, please? For `DocumentQueryResponse`, `documentQueryResponseSchema`, and `documentQueryRequestSchema`, I only see the schemas and types being declared and not used. `DocumentQueryRequest` I see it being used as the type of `getDocuments()` inbound param - but if the schema is build only for type definition, we can declare those with regular Typescript code and get rid of the Zod schema, much simpler.

Reply:
Gotcha so ill update the request because we wont be doing any parsing but the responses are going to be used by the api to parse the incoming payloads from Mirth (not implemented yet)

Reply:
Sounds good. Yeah, no worries about using zod schemas, just wanted to make sure we were going to use them, as they add some complexity in comparison to bare type definition. 👍